pyccel/ast/basic.py

```diff
@@ -165,7 +165,92 @@ def get_attribute_nodes(self, search_type, excluded_nodes = ()):
         self._recursion_in_progress = False
         return results
 
-    def substitute(self, original, replacement, excluded_nodes = ()):
+    def is_attribute_of(self, search_node, excluded_nodes = ()):
+        """ Identifies whether the object search_node is an attribute of
+        this object or one of its attributes
```

Comment:
I think that the two objects in this sentence should be swapped:
```suggestion
    def is_attribute_of(self, search_node, excluded_nodes = ()):
        """ Identifies whether this object is an attribute of search_node.
        The function searches recursively down the attribute tree.
```